### PR TITLE
Bp add create questionnaire code to v2

### PR DESCRIPTION
**Pull request by BarryPiccinni**
Add owner data to create questionnaire code for openapi V2.

### CODE REVIEW

File: questionnaire/questionnaire-service.js [hunk 1]
```diff
@@ -47,6 +49,25 @@ function createQuestionnaireService({
...
+                        name: 'OwnerNotFound'
+                    },
+                    `Owner data must be defined`
+                );
+            }
+
+            questionnaire.answers = {
+                ...questionnaire.answers,
+                owner: {
+                    'owner-id': owner.id,
```
Comment (sinclairs) [inline questionnaire/questionnaire-service.js:65]: I know we had a chat around the hyphen stuff... is it worth reverting to `ownerId` when storing it on the questionnaire?

File: questionnaire/questionnaire-service.js [hunk 2]
```diff
@@ -47,6 +49,25 @@ function createQuestionnaireService({
...
+                throw new VError(
+                    {
+                        name: 'OwnerNotFound'
+                    },
+                    `Owner data must be defined`
+                );
+            }
+
+            questionnaire.answers = {
+                ...questionnaire.answers,
```
Comment (sinclairs) [inline questionnaire/questionnaire-service.js:63]: This is the first ever answers in the questionnaire lifecycle, so we can drop the spread

**Review by armoj — CHANGES_REQUESTED** [review]

File: questionnaire/questionnaire-service.js [hunk 3]
```diff
@@ -20,6 +20,8 @@ defaults.createQuestionnaireDAL = require('./questionnaire-dal');
 
 function createQuestionnaireService({
     logger,
+    ownerId, // eslint-disable-line no-unused-vars
```
Comment (armoj) [inline questionnaire/questionnaire-service.js:23]: remove this. Add it in when it is relevant

File: questionnaire/questionnaire-service.test.js [hunk 4]
```diff
@@ -1,7 +1,27 @@
+/* eslint-disable no-shadow */
```
Comment (armoj) [inline questionnaire/questionnaire-service.test.js:1]: Are there Supertest routes test for the Express routing that implements these functions?

File: questionnaire/questionnaire-service.test.js [hunk 5]
```diff
@@ -192,26 +212,25 @@ function getQuestionnaireDefinition() {
...
-            );
-        }
-    })
-});
-
 describe('Questionnaire Service', () => {
     describe('Answering a section', () => {
+        const questionnaireService2 = createQuestionnaireService({
+            logger: () => 'Logged from dataset test',
+            createQuestionnaireDAL: () => ({
```
Comment (armoj) [inline questionnaire/questionnaire-service.test.js:219]: Can this be Mocked and just naturally passed in via the dependency injection? Not explicitly passed in to the `createQuestionnaireService` function

**Review by armoj — CHANGES_REQUESTED** [review]

File: questionnaire/questionnaire-service.test.js (hunk 5 above)
Comment (BarryPiccinni) [reply questionnaire/questionnaire-service.test.js:219]: I didn't change this, only moved it. I'd rather avoid refactoring old tests where I can as these will be deprecated with v1.

File: questionnaire/routes.js [hunk 6]
```diff
@@ -27,10 +27,13 @@ router.route('/').post(permissions('create:questionnaires'), async (req, res, ne
...
         }
 
-        const {templateName} = req.body.data.attributes;
+        const {templateName, owner} = req.body.data.attributes;
 
-        const questionnaireService = createQuestionnaireService({logger: req.log});
-        const response = await questionnaireService.createQuestionnaire(templateName);
+        const questionnaireService = createQuestionnaireService({
+            logger: req.log,
+            apiVersion: req.get['Dcs-Api-Version']
```
Comment (armoj) [inline questionnaire/routes.js:34]: Add a routing-level test to validate the header `Dcs-Api-Version` property

**Review by armoj — CHANGES_REQUESTED** [review]

**Pull request merged**